dart/test/sentry_event_test.dart

```diff
@@ -109,9 +108,6 @@ void main() {
             'params': ['1', '2']
           },
           'transaction': '/test/1',
-          'exception': [
```

Comment:
don't we test this anymore?

Reply:
it depends. `SentryEvent` has a `throwable`, an  `exception`, and a `stacktrace` fields.
The "normal" use case is to set a `throwable`, which will be converted to SentryException before transport.
Is there a use case in which the user set directly a SentryException ? in which case we need to add more verifications before setting the `event.exception` in `client.prepareEvent()`

Reply:
yes, a user can set its own `event.exception`, the thing is, on Java, we add an exception if already exist, because its a list, but in Dart is different, we only process the `throwable` if the given `exception` is null, so we dont overwrite a custom exception from the user.

https://github.com/getsentry/sentry-java/blob/113af8211c07ef622a4e265cc6b3438a6194d9fe/sentry/src/main/java/io/sentry/MainEventProcessor.java#L53-L56

this check is missing on Java